simpleline/render/widget_containers.py

```diff
@@ -0,0 +1,349 @@
+# Widgets for holding other widgets with possibility of automatic user
+# processing and numbering.
+#
+# Copyright (C) 2016  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from math import ceil
+
+from simpleline.render.widgets import Widget, TextWidget
+
+__all__ = ["ListRowWidget", "ListColumnWidget", "WidgetContainer", "KeyPattern"]
```

Comment:
It would be great if `self.window` in spokes was also a widget (or a container, see below). For example `WindowContainer`.